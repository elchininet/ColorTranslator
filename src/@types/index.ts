export type NumberOrString = number | string;

export interface HEXObject {
    r: string;
    g: string;
    b: string;
    a?: string;
}

export interface RGBObject {
    r: number;
    g: number;
    b: number;
    a?: number;
}

export interface RGBObjectGeneric {
    r: NumberOrString;
    g: NumberOrString;
    b: NumberOrString;
    a?: NumberOrString;
}

export interface HSLObject {
    h: number;
    s: number;
    l: number;
    a?: number;
}

export interface HSLObjectGeneric {
    h: number;
    s: NumberOrString;
    l: NumberOrString;
    a?: number;
}

export interface CMYKObject {
    c: number;
    m: number;
    y: number;
    k: number;
    a?: number;
}

export interface CMYKObjectGeneric {
    c: NumberOrString;
    m: NumberOrString;
    y: NumberOrString;
    k: NumberOrString;
    a?: number;
}

export interface RYBObject {
    r: number;
    y: number;
    b: number;
    a?: number;
}

export type RGYBObject = RGBObject | RYBObject;

export type Color = RGBObjectGeneric | HSLObjectGeneric | CMYKObjectGeneric;
export type ColorWithoutCMYK = RGBObjectGeneric | HSLObjectGeneric;

export type ColorInput = string | Color;
export type ColorInputWithoutCMYK = string | ColorWithoutCMYK;
export type HEXOutput = string | HEXObject;
export type RGBOutput = string | RGBObject;
export type HSLOutput = string | HSLObject;
export type CMYKOutput = string | CMYKObject;
export type ColorOutput = HEXOutput | RGBOutput | HSLOutput;

export interface ObjectProps<T> {
    [key: string]: T;
}

export interface Options {
    decimals: number;
    legacyCSS: boolean;
    spacesAfterCommas: boolean;
}

export type InputOptions = Partial<Options>;