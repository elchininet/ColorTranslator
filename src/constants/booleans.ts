export const DEFAULT_COLOR_LEVEL_4 = false;