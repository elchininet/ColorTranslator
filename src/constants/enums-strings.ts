import { Harmony, Mix } from './enums';

export type HarmonyString = `${Harmony}`;
export type MixString = `${Mix}`;