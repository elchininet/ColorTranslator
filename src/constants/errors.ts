export const ERRORS = {
    NOT_ACCEPTED_STRING_INPUT: 'The provided string color doesn\'t have a correct format',
    NOT_ACCEPTED_OBJECT_INPUT: 'The provided color object doesn\'t have the proper keys or format',
    NOT_A_VALID_RELATIVE_COLOR: 'The provided relative color is not valid'
};