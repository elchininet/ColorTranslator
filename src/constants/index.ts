export * from './numbers';
export * from './enums';
export * from './regexps';
export * from './errors';
