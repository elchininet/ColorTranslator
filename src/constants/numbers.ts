export const MAX_DECIMALS = 6;
export const MIN_DECIMALS = 2;
