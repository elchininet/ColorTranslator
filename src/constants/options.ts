import {
    Options,
    AnglesUnitEnum,
    ColorUnitEnum,
    CMYKFunctionEnum
} from '@types';
import { MAX_DECIMALS } from './numbers';

export const DEFAULT_OPTIONS: Options = {
    decimals: MAX_DECIMALS,
    legacyCSS: false,
    spacesAfterCommas: false,
    anglesUnit: AnglesUnitEnum.NONE,
    rgbUnit: ColorUnitEnum.NONE,
    labUnit: ColorUnitEnum.NONE,
    cmykUnit: ColorUnitEnum.PERCENT,
    alphaUnit: ColorUnitEnum.NONE,
    cmykFunction: CMYKFunctionEnum.DEVICE_CMYK
};