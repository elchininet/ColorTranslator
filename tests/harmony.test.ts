import { ColorTranslator } from '../src';
import { Harmony, Mix } from '../src/constants';
import { HEXObject, RGBObject, HSLObjectGeneric } from '../src/@types';

describe('ColorTranslator harmony tests', (): void => {

    const hexColorFunctions = [
        ColorTranslator.toHEX,
        ColorTranslator.toHEXA,
    ];

    const colorFunctions = [
        ColorTranslator.toRGB,
        ColorTranslator.toRGBA,
        ColorTranslator.toHSL,
        ColorTranslator.toHSLA
    ];

    const base = '#FF0000';

    const additive_results = [
        ['#FF0000', '#FF8000', '#FF0080'],
        ['#FF0000', '#00FFFF'],
        ['#FF0000', '#00FF80', '#0080FF'],
        ['#FF0000', '#00FF00', '#0000FF'],
        ['#FF0000', '#FFFF00', '#0000FF', '#00FFFF'],
        ['#FF0000', '#80FF00', '#8000FF', '#00FFFF']
    ];

    const subtractive_results = [
        ['#FF0000', '#FF4000', '#FF0080'],
        ['#FF0000', '#00FF00'],
        ['#FF0000', '#80FF00', '#00FFFF'],
        ['#FF0000', '#FFFF00', '#0000FF'],
        ['#FF0000', '#FF8000', '#0000FF', '#00FF00'],
        ['#FF0000', '#FFBF00', '#8000FF', '#00FF00']
    ];

    Object.keys(Harmony).forEach((harmony, index): void => {

        hexColorFunctions.forEach((fn): void => {

            const css = fn(base);
            const obj = fn(base, false) as HEXObject & RGBObject & HSLObjectGeneric;
            const additiveResultCSS = additive_results.map(colors => colors.map(color => fn(color)));
            const additiveResultObject = additive_results.map(colors => colors.map(color => fn(color, false))) as (HEXObject | RGBObject | HSLObjectGeneric)[][];
            const subtractiveResultCSS = subtractive_results.map(colors => colors.map(color => fn(color)));
            const subtractiveResultObject = subtractive_results.map(colors => colors.map(color => fn(color, false))) as (HEXObject | RGBObject | HSLObjectGeneric)[][];

            it(`Additive Harmony deep equals: ${harmony} for ${css} => ${JSON.stringify(additiveResultCSS[index])}`, (): void => {
                const colors = harmony === Harmony.COMPLEMENTARY
                    ? ColorTranslator.getHarmony(css)
                    : ColorTranslator.getHarmony(css, Harmony[harmony as Harmony]);
                expect(colors).toMatchObject(additiveResultCSS[index]);
            });

            it(`Additive Harmony deep equals: ${harmony} for ${JSON.stringify(obj)} => ${JSON.stringify(additiveResultObject[index])}`, (): void => {
                const colors = harmony === Harmony.COMPLEMENTARY
                    ? ColorTranslator.getHarmony(obj)
                    : ColorTranslator.getHarmony(obj, Harmony[harmony as Harmony]);
                expect(colors).toMatchObject(additiveResultObject[index]);
            });

            it(`Subtractive Harmony deep equals: ${harmony} for ${css} => ${JSON.stringify(subtractiveResultCSS[index])}`, (): void => {
                const colors = ColorTranslator.getHarmony(css, Harmony[harmony as Harmony], Mix.SUBTRACTIVE);
                expect(colors).toMatchObject(subtractiveResultCSS[index]);
            });

            it(`Subtractive Harmony deep equals: ${harmony} for ${JSON.stringify(obj)} => ${JSON.stringify(subtractiveResultObject[index])}`, (): void => {
                const colors = ColorTranslator.getHarmony(obj, Harmony[harmony as Harmony], Mix.SUBTRACTIVE);
                expect(colors).toMatchObject(subtractiveResultObject[index]);
            });

            it('Additive Complementary > 360ยบ', (): void => {
                const colors = ColorTranslator.getHarmony('#0000FF', Harmony.COMPLEMENTARY, Mix.SUBTRACTIVE);
                expect(colors[1]).toBe('#FF8000');
            });

        });

        colorFunctions.forEach((fn): void => {

            const css = fn(base, true, 0);
            const obj = fn(base, false, 0) as HEXObject & RGBObject & HSLObjectGeneric;
            const additiveResultCSS = additive_results.map(colors => colors.map(color => fn(color, true, 0)));
            const additiveResultObject = additive_results.map(colors => colors.map(color => fn(color, false, 0))) as (HEXObject | RGBObject | HSLObjectGeneric)[][];
            const subtractiveResultCSS = subtractive_results.map(colors => colors.map(color => fn(color, true, 0)));
            const subtractiveResultObject = subtractive_results.map(colors => colors.map(color => fn(color, false, 0))) as (HEXObject | RGBObject | HSLObjectGeneric)[][];

            it(`Additive Harmony deep equals: ${harmony} for ${css} => ${JSON.stringify(additiveResultCSS[index])}`, (): void => {
                const colors = ColorTranslator.getHarmony(css, Harmony[harmony as Harmony], Mix.ADDITIVE, 0);
                expect(colors).toMatchObject(additiveResultCSS[index]);
            });

            it(`Additive Harmony deep equals: ${harmony} for ${JSON.stringify(obj)} => ${JSON.stringify(additiveResultObject[index])}`, (): void => {
                const colors = ColorTranslator.getHarmony(obj, Harmony[harmony as Harmony], Mix.ADDITIVE, 0);
                expect(colors).toMatchObject(additiveResultObject[index]);
            });

            it(`Additive Harmony with decimals for ${JSON.stringify(css)}`, (): void => {
                const colors = ColorTranslator.getHarmony(css, Harmony[harmony as Harmony], Mix.ADDITIVE);
                expect(colors).toMatchSnapshot();
            });

            it(`Subtractive Harmony deep equals: ${harmony} for ${css} => ${JSON.stringify(subtractiveResultCSS[index])}`, (): void => {
                const colors = ColorTranslator.getHarmony(css, Harmony[harmony as Harmony], Mix.SUBTRACTIVE, 0);
                expect(colors).toMatchObject(subtractiveResultCSS[index]);
            });

            it(`Subtractive Harmony deep equals: ${harmony} for ${JSON.stringify(obj)} => ${JSON.stringify(subtractiveResultObject[index])}`, (): void => {
                const colors = ColorTranslator.getHarmony(obj, Harmony[harmony as Harmony], Mix.SUBTRACTIVE, 0);
                expect(colors).toMatchObject(subtractiveResultObject[index]);
            });

            it(`Subtractive Harmony with decimals for ${JSON.stringify(css)}`, (): void => {
                const colors = ColorTranslator.getHarmony(css, Harmony[harmony as Harmony], Mix.SUBTRACTIVE);
                expect(colors).toMatchSnapshot();
            });

            it('Additive Complementary > 360ยบ', (): void => {
                const colors = ColorTranslator.getHarmony('#0000FF', Harmony.COMPLEMENTARY, Mix.SUBTRACTIVE, 0);
                expect(colors[1]).toBe('#FF8000');
            });

        });

    });

});