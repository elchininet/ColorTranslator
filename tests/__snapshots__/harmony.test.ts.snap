// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsl(0, 100%, 50%)" 1`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(30, 100%, 50%)",
  "hsl(330, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsl(0, 100%, 50%)" 2`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(180, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsl(0, 100%, 50%)" 3`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(150, 100%, 50%)",
  "hsl(210, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsl(0, 100%, 50%)" 4`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(120, 100%, 50%)",
  "hsl(240, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsl(0, 100%, 50%)" 5`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(60, 100%, 50%)",
  "hsl(240, 100%, 50%)",
  "hsl(180, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsl(0, 100%, 50%)" 6`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(90, 100%, 50%)",
  "hsl(270, 100%, 50%)",
  "hsl(180, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 1`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(30, 100%, 50%, 1)",
  "hsla(330, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 2`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(180, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 3`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(150, 100%, 50%, 1)",
  "hsla(210, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 4`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(120, 100%, 50%, 1)",
  "hsla(240, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 5`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(60, 100%, 50%, 1)",
  "hsla(240, 100%, 50%, 1)",
  "hsla(180, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 6`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(90, 100%, 50%, 1)",
  "hsla(270, 100%, 50%, 1)",
  "hsla(180, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgb(255, 0, 0)" 1`] = `
[
  "rgb(255, 0, 0)",
  "rgb(255, 127.5, 0)",
  "rgb(255, 0, 127.5)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgb(255, 0, 0)" 2`] = `
[
  "rgb(255, 0, 0)",
  "rgb(0, 255, 255)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgb(255, 0, 0)" 3`] = `
[
  "rgb(255, 0, 0)",
  "rgb(0, 255, 127.5)",
  "rgb(0, 127.5, 255)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgb(255, 0, 0)" 4`] = `
[
  "rgb(255, 0, 0)",
  "rgb(0, 255, 0)",
  "rgb(0, 0, 255)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgb(255, 0, 0)" 5`] = `
[
  "rgb(255, 0, 0)",
  "rgb(255, 255, 0)",
  "rgb(0, 0, 255)",
  "rgb(0, 255, 255)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgb(255, 0, 0)" 6`] = `
[
  "rgb(255, 0, 0)",
  "rgb(127.5, 255, 0)",
  "rgb(127.5, 0, 255)",
  "rgb(0, 255, 255)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgba(255, 0, 0, 1)" 1`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(255, 127.5, 0, 1)",
  "rgba(255, 0, 127.5, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgba(255, 0, 0, 1)" 2`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(0, 255, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgba(255, 0, 0, 1)" 3`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(0, 255, 127.5, 1)",
  "rgba(0, 127.5, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgba(255, 0, 0, 1)" 4`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(0, 255, 0, 1)",
  "rgba(0, 0, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgba(255, 0, 0, 1)" 5`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(255, 255, 0, 1)",
  "rgba(0, 0, 255, 1)",
  "rgba(0, 255, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Additive Harmony with decimals for "rgba(255, 0, 0, 1)" 6`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(127.5, 255, 0, 1)",
  "rgba(127.5, 0, 255, 1)",
  "rgba(0, 255, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsl(0, 100%, 50%)" 1`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(15, 100%, 50%)",
  "hsl(330, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsl(0, 100%, 50%)" 2`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(120, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsl(0, 100%, 50%)" 3`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(90, 100%, 50%)",
  "hsl(180, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsl(0, 100%, 50%)" 4`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(60, 100%, 50%)",
  "hsl(240, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsl(0, 100%, 50%)" 5`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(30, 100%, 50%)",
  "hsl(240, 100%, 50%)",
  "hsl(120, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsl(0, 100%, 50%)" 6`] = `
[
  "hsl(0, 100%, 50%)",
  "hsl(45, 100%, 50%)",
  "hsl(270, 100%, 50%)",
  "hsl(120, 100%, 50%)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 1`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(15, 100%, 50%, 1)",
  "hsla(330, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 2`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(120, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 3`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(90, 100%, 50%, 1)",
  "hsla(180, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 4`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(60, 100%, 50%, 1)",
  "hsla(240, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 5`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(30, 100%, 50%, 1)",
  "hsla(240, 100%, 50%, 1)",
  "hsla(120, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "hsla(0, 100%, 50%, 1)" 6`] = `
[
  "hsla(0, 100%, 50%, 1)",
  "hsla(45, 100%, 50%, 1)",
  "hsla(270, 100%, 50%, 1)",
  "hsla(120, 100%, 50%, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgb(255, 0, 0)" 1`] = `
[
  "rgb(255, 0, 0)",
  "rgb(255, 63.75, 0)",
  "rgb(255, 0, 127.5)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgb(255, 0, 0)" 2`] = `
[
  "rgb(255, 0, 0)",
  "rgb(0, 255, 0)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgb(255, 0, 0)" 3`] = `
[
  "rgb(255, 0, 0)",
  "rgb(127.5, 255, 0)",
  "rgb(0, 255, 255)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgb(255, 0, 0)" 4`] = `
[
  "rgb(255, 0, 0)",
  "rgb(255, 255, 0)",
  "rgb(0, 0, 255)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgb(255, 0, 0)" 5`] = `
[
  "rgb(255, 0, 0)",
  "rgb(255, 127.5, 0)",
  "rgb(0, 0, 255)",
  "rgb(0, 255, 0)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgb(255, 0, 0)" 6`] = `
[
  "rgb(255, 0, 0)",
  "rgb(255, 191.25, 0)",
  "rgb(127.5, 0, 255)",
  "rgb(0, 255, 0)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgba(255, 0, 0, 1)" 1`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(255, 63.75, 0, 1)",
  "rgba(255, 0, 127.5, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgba(255, 0, 0, 1)" 2`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(0, 255, 0, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgba(255, 0, 0, 1)" 3`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(127.5, 255, 0, 1)",
  "rgba(0, 255, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgba(255, 0, 0, 1)" 4`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(255, 255, 0, 1)",
  "rgba(0, 0, 255, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgba(255, 0, 0, 1)" 5`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(255, 127.5, 0, 1)",
  "rgba(0, 0, 255, 1)",
  "rgba(0, 255, 0, 1)",
]
`;

exports[`ColorTranslator harmony tests Subtractive Harmony with decimals for "rgba(255, 0, 0, 1)" 6`] = `
[
  "rgba(255, 0, 0, 1)",
  "rgba(255, 191.25, 0, 1)",
  "rgba(127.5, 0, 255, 1)",
  "rgba(0, 255, 0, 1)",
]
`;
